README.adoc

```diff
@@ -1,43 +1,186 @@
 ----
-List available Kamelet sources
+
+=== `describe-type`
+
+----
+Show details of given Kamelet source type
 
 Usage:
-  kn-source-kamelet list [flags]
+  kn-source-kamelet describe-type [flags]
+
+Aliases:
+  describe-type, dt
+
+Examples:
+
+  # Describe given Kamelets
+  kn-source-kamelet describe-type NAME
+
+  # Describe given Kamelets in YAML output format
+  kn-source-kamelet describe-type NAME -o yaml
+
+Flags:
+      --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
+  -h, --help                          help for describe-type
+  -n, --namespace string              Specify the namespace to operate in.
+  -o, --output string                 Output format. One of: json|yaml|name|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file|url.
+      --show-managed-fields           If true, keep the managedFields when printing objects in JSON or YAML format.
+      --template string               Template string or path to template file to use when -o=go-template, -o=go-template-file. The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].
+  -v, --verbose                       More output.
+----
+
+=== `binding`
+
+----
+Configure and manage a Kamelet binding.
+
+Usage:
+  kn-source-kamelet binding [command]
+
+Examples:
+
+  # Configure and manage a Kamelet binding.
+  kn-source-kamelet binding create|update|delete
+
+Available Commands:
+  create      Create Kamelet bindings and bind source to Knative broker, channel or service.
+
+Flags:
+  -h, --help   help for binding
+
+Use "kn-source-kamelet binding [command] --help" for more information about a command.
+----
+
+==== `binding create`
+
+----
+Create Kamelet bindings and bind source to Knative broker, channel or service.
+
+Usage:
+  kn-source-kamelet binding create [flags]
+
+Examples:
+
+  # Create Kamelet binding with source and sink.
+  kn-source-kamelet binding create NAME
+
+  # Add a binding properties
+  kn-source-kamelet binding create NAME --kamelet=name --sink|broker|channel|service=<name> --source-property=<key>=<value> --sink-property=<key>=<value>
+
+Flags:
+      --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
+      --broker string                 Uses a broker as binding sink.
+      --channel string                Uses a channel as binding sink.
+  -h, --help                          help for create
+      --kamelet string                Kamelet source.
+  -n, --namespace string              Specify the namespace to operate in.
+  -o, --output string                 Output format. One of: json|yaml|name|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file|url.
```

Comment:
What do you output with `create` ?

Reply:
the created binding resource as it has been created (for debug reasons)

Reply:
I would prefer to keep the output options for a `binding describe` as we have for all other commands and plugins. It is quite confusing to see those options for output on a `create` or `bind` command (it's more than one option).

The convention for kn is to print out a human-readable description of what has been achieved. For `bind` I would highly recommend printing out the name of the created binding as this is auto generated.

Reply:
btw, I would mind if you print out the YAML / JSON file when a `--debug` option is given, if you would need if for debugging purposes. Also, you could think about an offline mode like we have for `kn service create --target` that would create the corresponding resource file without applying it to a cluster (good for gitops).